src/lib.js

```diff
@@ -1704,11 +1703,14 @@ export const getStats = async (
       output.stats[stat] += fairyBonus[stat];
     }
   }
+  const totalSouls = profile.game_mode === "island"
+    ? constants.MAX_SOULS.stranded
+    : constants.MAX_SOULS.normal;
 
   output.fairy_souls = {
     collected: userProfile.fairy_souls_collected,
-    total: MAX_SOULS,
-    progress: Math.min(userProfile.fairy_souls_collected / MAX_SOULS, 1),
+    total: totalSouls,
+    progress: (userProfile.fairy_souls_collected / totalSouls),
```

Comment:
Looks good but it's not formatted (prettier)... the parenthesis here are useless